.github/actions/setup_bazel_nix/action.yml

```diff
@@ -267,3 +271,42 @@ runs:
         echo "common --disk_cache=" >> "${WORKSPACE}/.bazeloverwriterc"
         echo "common --repository_cache=" >> "${WORKSPACE}/.bazeloverwriterc"
         echo "::endgroup::"
+
+    - name: Install nix tools
+      if: inputs.nixTools != ''
+      shell: bash
+      env:
+        tools: ${{ inputs.nixTools }}
+        repository: ${{ github.repository }}
+        gitSha: ${{ github.sha }}
+      run: |
+        echo "::group::Install nix tools"
+        expressionFile=$(mktemp)
+        cat << "EOF" > "${expressionFile}"
+        { tools, repository, rev }:
+        let
+          repoFlake = builtins.getFlake ("github:" + repository + "/" + rev);
+          nixpkgs = repoFlake.inputs.nixpkgsUnstable;
+          pkgs = import nixpkgs { system = builtins.currentSystem; };
+          toolPkgs = map (p: pkgs.${p}) tools;
+        in
+        {
+          tools = pkgs.symlinkJoin { name = "tools"; paths = [ toolPkgs ]; };
+          pathVar = pkgs.lib.makeBinPath toolPkgs;
+        }
+        EOF
+        nix-build \
+          --no-out-link \
+          --arg tools "${tools}" \
+          --argstr repository "${repository}" \
+          --argstr rev "${gitSha}" \
+          --attr tools \
+          "${expressionFile}"
+        EXTRA_PATH=$(nix eval --raw --file "${expressionFile}" \
+          --arg tools "${tools}" \
+          --argstr repository "${repository}" \
+          --argstr rev "${gitSha}" \
+          pathVar)
+        echo "EXTRA_PATH=${EXTRA_PATH}"
+        echo "${EXTRA_PATH}" >> "${GITHUB_PATH}"
```

Comment:
If a tool isn't installed by us, it will still fall back to the preinstalled tools, right?

Reply:
We should investigate if there is a way to clear the path and ensure only explicitly declared tools are used, but not as part of this PR.